tests/regression/test_all_round_happy_path.py

```diff
@@ -0,0 +1,173 @@
+from brownie import ZERO_ADDRESS
+
+from utils.test.oracle_report_helpers import oracle_report
+from utils.test.helpers import ETH, almostEqEth
+from utils.config import contracts
+
+
+def test_all_round_happy_path(accounts):
+    stranger = accounts[0]
+    amount = ETH(100)
+    max_deposit = 150
+    curated_module_id = 1
+
+    steth_balance_before_submit = contracts.lido.balanceOf(stranger)
+    eth_balance_before_submit = stranger.balance()
+
+    assert steth_balance_before_submit == 0
+
+    # Submitting ETH
+
+    total_supply_before_submit = contracts.lido.totalSupply()
+    buffered_ether_before_submit = contracts.lido.getBufferedEther()
+    staking_limit_before_submit = contracts.lido.getCurrentStakeLimit()
+
+    submit_tx = contracts.lido.submit(ZERO_ADDRESS, {"from": stranger, "amount": amount})
+
+    steth_balance_after_submit = contracts.lido.balanceOf(stranger)
+    total_supply_after_submit = contracts.lido.totalSupply()
+    buffered_ether_after_submit = contracts.lido.getBufferedEther()
+    staking_limit_after_submit = contracts.lido.getCurrentStakeLimit()
+
+    assert almostEqEth(steth_balance_after_submit, steth_balance_before_submit + amount)
+    assert eth_balance_before_submit == stranger.balance() + amount
+
+    submit_event = submit_tx.events["Submitted"]
+    transfer_shares_event = submit_tx.events["TransferShares"]
+
+    assert submit_event["sender"] == stranger
+    assert submit_event["amount"] == amount
+    assert submit_event["referral"] == ZERO_ADDRESS
+
+    shares_to_be_minted = contracts.lido.getSharesByPooledEth(amount)
+
+    assert transfer_shares_event["from"] == ZERO_ADDRESS
+    assert transfer_shares_event["to"] == stranger
+    assert almostEqEth(transfer_shares_event["sharesValue"], shares_to_be_minted)
+
+    assert total_supply_after_submit == total_supply_before_submit + amount
+    assert buffered_ether_after_submit == buffered_ether_before_submit + amount
+    assert staking_limit_after_submit == staking_limit_before_submit - amount
+
+    # Depositing ETH
+    dsm = accounts.at(contracts.deposit_security_module.address, force=True)
+
+    contracts.lido.deposit(max_deposit, curated_module_id, "0x0", {"from": dsm})
```

Comment:
can add a check for `getDepositableEther` (should be equal to `getBufferedEther` when no withdrawal request placed)